### ISSUE TITLE
关于loss的计算

### ISSUE THREAD

**Issue by Y1YU** [BR-IDL/PaddleViT#104]
交叉熵loss应该计算预测分布概率与真实分布之间的距离，按道理是应该在softmax计算之后再进行Loss计算的，但是训练代码中都是直接计算loss `loss=criterion(output, label)`，后面才再计算softmax。请问不应该是先计算softmax再计算Loss，还是其实这两者的顺序对训练没有区别

**Comment by libertatis** [comment]
其实 `softmax` 操作是在损失函数里面做了的。这里的交叉熵损失函数使用的是 `PaddleViT` 自定义的 `CrossEntropyLoss` （[点积查看源码](https://github.com/BR-IDL/PaddleViT/blob/796043557bb617abb22fa546876000cc0c8946e4/semantic_segmentation/src/models/losses/cross_entropy_loss.py#L20)），它是通过调用 `paddle.nn.functional.cross_entropy` 函数实现的，而 `cross_entropy` 函数则将 `softmax` 操作和交叉熵的计算算进行了合并（[点积查看文档](https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/nn/functional/cross_entropy_cn.html)）。

**Comment by libertatis** [comment]
其实 `softmax` 操作是在损失函数里面做了的。这里的交叉熵损失函数使用的是 `PaddlePaddle` 自带的 `paddle.nn.CrossEntropyLoss` （[点积查看源码](https://github.com/PaddlePaddle/Paddle/blob/release/2.2/python/paddle/nn/layer/loss.py#L132)） 或是 `PaddleViT` 自定义的 `CrossEntropyLoss` （[点积查看源码](https://github.com/BR-IDL/PaddleViT/blob/796043557bb617abb22fa546876000cc0c8946e4/semantic_segmentation/src/models/losses/cross_entropy_loss.py#L20)），它们是通过调用 `paddle.nn.functional.cross_entropy` 函数实现的，而 `cross_entropy` 函数则将 `softmax` 操作和交叉熵的计算算进行了合并（[点积查看文档](https://www.paddlepaddle.org.cn/documentation/docs/zh/api/paddle/nn/functional/cross_entropy_cn.html)）。

**Comment by xperzy** [comment]
@libertatis Thanks for the clear explanations.  I'm closing this issue.
